File: web/src/main/vite-project/src/kostratypes/fileReportVm.ts
import KostraFormVm from "./kostraFormVm"
import FileReportEntryVm from "./fileReportEntryVm"
import KostraSeverity from "./kostraSeverity"
import Nullable from "./nullable";

interface FileReportVm {
    readonly innparametere: NonNullable<KostraFormVm>
    readonly antallKontroller: NonNullable<number>
    readonly feilkode: NonNullable<KostraSeverity>
    readonly feil: Nullable<FileReportEntryVm[]>
}

export default FileReportVm

File: application-common/src/main/kotlin/no/ssb/kostra/common/viewmodel/FileReportVm.kt
package no.ssb.kostra.common.viewmodel

import io.micronaut.core.annotation.Introspected

@Introspected
data class FileReportVm(
    val innparametere: KostraFormVm,
    val antallKontroller: Int,
    val feilkode: KostraErrorCode,
    val feil: Collection<FileReportEntryVm>
)
